src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+
+void EnWf_BackFlip(EnWf* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (!Actor_OtherIsTargeted(globalCtx, &this->actor) && (this->actor.xzDistToPlayer < 170.0f) &&
+            (this->actor.xzDistToPlayer > 140.0f) && (Rand_ZeroOne() < 0.2f)) {
+            EnWf_SetupRunAtPlayer(this, globalCtx);
+        } else if ((globalCtx->gameplayFrames % 2) != 0) {
+            EnWf_SetupSidestep(this, globalCtx);
+        } else {
+            EnWf_SetupWait(this);
+        }
+    }
+    if ((globalCtx->state.frames & 95) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_CRY);
+    }
+}
+
+void EnWf_SetupStunned(EnWf* this) {
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    Animation_PlayOnceSetSpeed(&this->skelAnime, &gWolfosKnockedbackAnim, 0.0f);
+    this->action = WOLFOS_ACTION_STUNNED;
+    EnWf_SetupAction(this, EnWf_Stunned);
+}
+
+void EnWf_Stunned(EnWf* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 2) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ += 0.05f;
+        }
+
+        this->isInvisible = false;
+    }
+
+    if ((this->actor.colorFilterTimer == 0) && (this->actor.bgCheckFlags & 1)) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnWf_SetupDie(this);
+        } else {
+            EnWf_ChangeAction(globalCtx, this, 1);
+        }
+    }
+}
+
+void EnWf_SetupDamaged(EnWf* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gWolfosKnockedbackAnim, -4.0f);
+
+    if (this->actor.bgCheckFlags & 1) {
+        this->isInvisible = false;
+        this->actor.speedXZ = -4.0f;
+    } else {
+        this->isInvisible = true;
+    }
+
+    this->headTilt = 0;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_DAMAGE);
+    this->action = WOLFOS_ACTION_DAMAGED;
+    EnWf_SetupAction(this, EnWf_Damaged);
+}
+
+void EnWf_Damaged(EnWf* this, GlobalContext* globalCtx) {
+    s16 angleToWall;
+
+    if (this->actor.bgCheckFlags & 2) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ += 0.05f;
+        }
+
+        this->isInvisible = false;
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 4500, 0);
+
+    if ((!(EnWf_ChangeAction(globalCtx, this, 0))) && SkelAnime_Update(&this->skelAnime)) {
+        if (this->actor.bgCheckFlags & 1) {
+            angleToWall = this->actor.wallYaw - this->actor.shape.rot.y;
+            angleToWall = ABS(angleToWall);
+
+            if (this->actor.bgCheckFlags & 8) {
+                if ((ABS(angleToWall) < 12000) && (this->actor.xzDistToPlayer < 120.0f)) {
```

Comment:
I don't see why not && these
if the concern is formatting, I would at least put `bgCheckFlags & 8` and `ABS(angleToWall) < 12000` in the same conditional

Reply:
You're right, and I can get rid of the early return as well.